File: backend/src/main/java/it/unical/inf/ea/backend/data/services/interfaces/BrandService.java
package it.unical.inf.ea.backend.data.services.interfaces;
import it.unical.inf.ea.backend.data.entities.Brand;
import it.unical.inf.ea.backend.dto.BrandDTO;
import it.unical.inf.ea.backend.dto.creation.BrandCreateDTO;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public interface BrandService{

    void addBrand(BrandCreateDTO brandDTO);
    void deleteBrand(Long id);
    List<BrandDTO> getAllBrands();
    Optional<Brand> findBrandById(Long id);
    Optional<Brand> findBrandByName(String brandName);
}

File: frontend/app/src/main/java/com/android/frontend/service/BrandService.kt
package com.android.frontend.service

import com.android.frontend.dto.BrandDTO
import com.android.frontend.dto.creation.BrandCreateDTO
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Query

interface BrandService {

    @POST("/api/v1/brand/addBrand")
    fun addBrand(
        @Header("Authorization") token: String,
        @Body brandCreateDTO: BrandCreateDTO
    ): Call<Void>

    @DELETE("/api/v1/brand/deleteBrand")
    fun deleteBrand(
        @Header("Authorization") token: String,
        @Query("id") id: Int
    ): Call<String>

    @GET("/api/v1/brand/allBrands")
    fun getAllBrands(
        @Header("Authorization") token: String,
    ): Call<List<BrandDTO>>
}
